cogeo_mosaic/backends/base.py

```diff
@@ -126,19 +125,25 @@ def update(
         self.bounds = bounds
         self.write(overwrite=True)
 
-    def assets_for_tile(self, x: int, y: int, z: int) -> List[str]:
-        """Retrieve assets for tile."""
-        return self.get_assets(x, y, z)
-
-    def assets_for_point(self, lng: float, lat: float) -> List[str]:
+    def assets_for_point(self, lng: float, lat: float, coord_crs: CRS = WGS84_CRS) -> List[str]:
         """Retrieve assets for point."""
+        lng, lat = transform_point(lng, lat, to_rio_crs(coord_crs), self.geographic_crs)
```

Comment:
🤔  I would assume `CRS` to always be in form of rasterio crs as input, so if there is a transformation to be made using the tms's CRS to rasterio's CRS.

If it helps we should add `rasterio_geographic_crs` in morecantile model.

having both pyproj and rasterio within the same module worries me!